### ISSUE TITLE
BNInception Training Problem

### ISSUE THREAD

**Issue by Prakashvanapalli** [Cadene/pretrained-models.pytorch#27]
Hi,

I tried training bninception on my own dataset but it is not working. May I know the problem.

Epoch 0/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:22<00:00,  2.94it/s]
train Loss: 0.0383 Acc: 0.1293
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.47it/s]
val Loss: 2.5915 Acc: 0.1079

Epoch 1/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:17<00:00,  3.84it/s]
train Loss: 0.0382 Acc: 0.1364
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.73it/s]
val Loss: 0.0425 Acc: 0.0809

Epoch 2/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:17<00:00,  3.80it/s]
train Loss: 0.0382 Acc: 0.1286
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.35it/s]
val Loss: 70.5504 Acc: 0.0560

Epoch 3/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:17<00:00,  3.74it/s]
train Loss: 0.0382 Acc: 0.1214
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.53it/s]
val Loss: 228.3415 Acc: 0.0560

Epoch 4/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:17<00:00,  3.80it/s]
train Loss: 0.0382 Acc: 0.1408
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.46it/s]
val Loss: 12274.5517 Acc: 0.1079

Epoch 5/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:18<00:00,  3.68it/s]
train Loss: 0.0383 Acc: 0.1202
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.00it/s]
val Loss: 74337.8683 Acc: 0.0560

Epoch 6/99
----------
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:17<00:00,  3.83it/s]
train Loss: 0.0382 Acc: 0.1228
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:02<00:00,  3.00it/s]
val Loss: 152427.3008 Acc: 0.0560

I have basically used my own dataloaders and training function. I trained nasnet and other networks available here and they are working fine, only bninception is not working. 

used batch_size=64, converted to BGR mode and Range255, also used mean and std as provided in the bninception.py file. 

My implementation is quite lengthy, so not producing here.

**Comment by Cadene** [comment]
@Prakashvanapalli Sorry for the late answer. I don't know for your problem. It works well for me.

Did you try this [code](https://github.com/Cadene/pretrained-models.pytorch/blob/master/examples/voc2007_extract.py) ? It is easier to try than imagenet.

```
CUDA_VISIBLE_DEVICES=0 python examples/voc2007_extract.py \
--dir_datasets /local/cadene/data/voc2007_extract \
--dir_outputs /tmp/voc2007_extract \
-a bninception \
--cuda \
--train_split train \
--test_split val
```

```
>   - train   : mAP 100.0000
  - val     : mAP 74.8318
```
